lisa/env.py

```diff
@@ -210,6 +212,46 @@ def from_one_conf(cls, path):
             plat_info = None
         return cls(target_conf=target_conf, plat_info=plat_info)
 
+    @classmethod
+    def from_cli(cls):
+        """
+        Create a TestEnv from command line arguments
+        """
+        description = textwrap.dedent(
+            """
+            Extra arguments differ depending on the value passed to 'kind'.
+            Try e.g. \"--kind android\" to see the arguments for android targets.
+            """
+        )
+        parser = argparse.ArgumentParser(description=description,
+                                         formatter_class=argparse.RawDescriptionHelpFormatter)
+        parser.add_argument("--kind", "-k",
+                            choices=["android", "linux", "host"],
+                            required=True,
+                            help="The kind of target to create")
+
+        # subparsers cannot let us specify --kind=android, at best we could
+        # have --android which is lousy.
+        # Use parse_known_args() instead to look at --kind, and add relevent
+        # arguments
+        args = parser.parse_known_args()[0]
+
+        if args.kind == "android":
+            parser.add_argument("--device", "-d", type=str, required=True,
+                                help="The ADB ID of the target")
+
+        elif args.kind == "linux":
+            parser.add_argument("--hostname", "-n", type=str, required=True, dest="host",
+                                help="The hostname/IP of the target")
+            parser.add_argument("--username", "-u", type=str, required=True,
+                                help="Login username")
+            parser.add_argument("--password", "-p",  type=str, required=True,
+                                help="Login password")
+
+        args = parser.parse_args()
+
+        return TestEnv(TargetConf(vars(args)))
```

Comment:
A --platform-info CLI option would allow creating a PlatformInfo from a file if specified. I don't think it's a good idea to try to build one from CLI options directly, since it is full of objects that are not strings, which means we would need to parse that ourselves. TargetConf is not so much of an issue since it only uses "simple" keys with YAML builtin basic types, and will not change much overtime.
`PlatformInfo.from_yaml_map(FILENAME)` can be used to build it if FILENAME is not None.